### PR TITLE
date fields added

### PR DESCRIPTION
**Pull request by provishalk**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ mongoose.connect(
     "mongodb+srv://vishal:12345@cluster0.nqdbc.mongodb.net/ManagementToolForHR?retryWrites=true&w=majority",
     { useNewUrlParser: true, useUnifiedTopology: true }
 );
-
+mongoose.set('useFindAndModify', false);
```
Comment (shivdotpy) [inline index.js:16]: resson of using this?

File: src/controllers/employee.controller.js [hunk 2]
```diff
@@ -23,18 +25,79 @@ module.exports.addEmployee = async (req,res) => {
             employementStatus: req.body.employementStatus,
             address: req.body.address,
             contact: contact,
+            joiningDate: date // Year/Month/Date
         });
-        newEmployee.save().then(() => console.log('1 Data Uploaded'));
+        console.log(newEmployee);
```
Comment (shivdotpy) [inline src/controllers/employee.controller.js:30]: Remove comments

File: index.js (hunk 1 above)
Comment (provishalk) [reply index.js:16]: I was getting Deprecation Warnings while using ( fineOne() ) that's why I have used this.

File: src/controllers/employee.controller.js [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
     const checkIsNumberRagex = /^[0-9]{10}$/; //Ragex Condition
-    
-
-    if (req.body.firstName.trim().length === 0 //Check Employee Name is empty or not
-        || req.body.lastName.trim().length === 0 //Check Employee Name is empty or not
-        || req.body.employementStatus.trim().length === 0 //Check Employee status is empty or not
-        || (await Employee.find({"contact":contact})).length != 0 //Check Employee already exist or not
-        || !checkIsNumberRagex.test(contact) //Check contact number is of length 10 and does not contain any string
-        
+    const date = new Date(req.body.joiningDate)
```
Comment (shivdotpy) [inline src/controllers/employee.controller.js:8]: did you add validation for date herE?

File: src/controllers/employee.controller.js [hunk 4]
```diff
@@ -23,18 +25,79 @@ module.exports.addEmployee = async (req,res) => {
...
+
+module.exports.updateEmployeeStatus = async (req, res) => {
+    const date = new Date(req.body.leavingDate)
+    if (req.body.employementStatus.trim().length === 0) {
         res.sendStatus(406);
     }
-    else{
-        let doc = await Employee.findOneAndUpdate({_id : req.body._id} , {employementStatus:req.body.employementStatus});
+    else {
+        if (req.body.employementStatus.toUpperCase() === "LEAVED") {
```
Comment (shivdotpy) [inline src/controllers/employee.controller.js:42]: move LEAVED to a constant file

File: src/controllers/employee.controller.js [hunk 5]
```diff
@@ -23,18 +25,79 @@ module.exports.addEmployee = async (req,res) => {
...
+            if (date == "Invalid Date") {
```
Comment (shivdotpy) [inline src/controllers/employee.controller.js:43]: move INVALID DATE to constant file

File: src/controllers/employee.controller.js [hunk 6]
```diff
@@ -23,18 +25,79 @@ module.exports.addEmployee = async (req,res) => {
...
+                await Employee.findOneAndUpdate({ _id: req.body._id }, { employementStatus: req.body.employementStatus, leavingDate: new Date() });
+            }
+            else {
+                await Employee.findOneAndUpdate({ _id: req.body._id }, { employementStatus: req.body.employementStatus, leavingDate: date });
```
Comment (shivdotpy) [inline src/controllers/employee.controller.js:47]: Need to refactor this code, not accepted

File: src/controllers/employee.controller.js [hunk 7]
```diff
@@ -23,18 +25,79 @@ module.exports.addEmployee = async (req,res) => {
...
+            {
+                $push: { dayLog: dayLog },
+                totalWorkingHours: newtotalWorkingHours
+            },
+            (err, result) => {
+                console.log(err)
+            })
+    }
+    else {
+        // console.log(timeIn.toLocaleTimeString());
```
Comment (shivdotpy) [inline src/controllers/employee.controller.js:91]: remove unwanted comments

File: src/controllers/employee.controller.js [hunk 8]
```diff
@@ -23,18 +25,79 @@ module.exports.addEmployee = async (req,res) => {
...
+        const firstTimeAttendance = new EmpAttendance(
+            {
+                employeeId: req.body._id,
+                dayLog: [{ timeIn: timeIn, timeOut: timeOut, totalTimeOfDay: totalHoursOfDay }],
+                totalWorkingHours: totalHoursOfDay
+            }
+        );
+        firstTimeAttendance.save().then(() => console.log('attendance Added'));
+    }
+    res.send("ok");
```
Comment (shivdotpy) [inline src/controllers/employee.controller.js:102]: use res.status(200).send({error: false, message: "ok"})

File: src/models/empAttendance.model.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require("mongoose");
+
+const EmpAttendanceModel = mongoose.Schema(
+    {
+        employeeId: String,
```
Comment (shivdotpy) [inline src/models/empAttendance.model.js:5]: Change it to flat model

**Pull request closed without merging**